### ISSUE TITLE
angular/ng_di: false positive for array constants

### ISSUE THREAD

**Issue by JamieMason** [Gillespie59/eslint-plugin-angular#99]
The following code is considered a violation of `angular/ng_di`

```javascript
(function() {

    'use strict';

    angular.module('SomeApp.constants', [])
        .config(configure);

    function configure($provide) {
        $provide.constant('SOME_LIST', ['foo', 'bar']);
    }

}());
```

```
src/app/constants.js
  9:8  error  You should use the function syntax for DI  angular/ng_di
```

Thanks

**Comment by Gillespie59** [comment]
Hi. 

I have just tried with the 0.0.8 branch and i do not have any problem.

In fact we so not support the config method in this rule. Are you sure that this snippet is the source of this error?

**Comment by Gillespie59** [comment]
OK. In fact the pb come from the constant method.  I am working on that right now

**Comment by Gillespie59** [comment]
I have just pushed a fix for this issue on the 0.0.8 branch. 
I am going to release the 0.0.8 version tonight I think.

**Comment by breandr** [comment]
I'm on 0.1.1 and getting this error on this:

```
myModule
.constant('API_URL_PREFIX', '/api')
.config(($httpProvider, $locationProvider, $sceDelegateProvider, $animateProvider) =>  {
....
```

The odd thing is that it's actually showing the error is on the `myModule` line

**Comment by Gillespie59** [comment]
@breandr  what is your ESLint configuration for the ng_di rule ?

**Comment by breandr** [comment]
"angular/ng_di": [2, 'function'],

**Comment by flipjs** [comment]
I am getting this error too from gulp-eslint when using the value service for arrays. Here's the code:

```
  angular
    .module('home')
    .value('myArray', [{
      name: 'fizz',
      email: 'fizz@example.com'
    }, {
      name: 'buzz',
      email: 'buzz@example.com'
    }]);
```

But when I change the service to constant (same code, just the literal word 'value' changed to 'constant'), no errors reported.

**Comment by devroyaldev** [comment]
Hello, I receive the message 'You should use the function syntax for DI' on that particular boilerplate code. What is wrong? 

'use strict';

// Declare app level module which depends on views, and components
angular.module('myApp', [
  'ngRoute',
  'myApp.view1',
  'myApp.view2',
  'myApp.version'
]).
config(['$routeProvider', function($routeProvider) {
  $routeProvider.otherwise({redirectTo: '/view1'});
}]);
